build errors with gcc 11 (atomic as volatile and loop construct)

The following template is for Issue/Bug reporting only. https://docs.qgroundcontrol.com/en/Support/Support.html#github-issues

For questions about how to use or build QGC see: http://qgroundcontrol.com/#resources

If you are using an old stable release please upgrade first and verify your issue before reporting.
----

## Expected Behavior
Please tell us what should happen as detailed as possible.
The compilation process should succeed.

## Current Behavior
Please describe what happens instead of the expected behavior.
QGroundControl fails to build.

## Steps to Reproduce:
Please provide an unambiguous set of steps to reproduce the current behavior
1. git clone ...
2. mkdir build
3. qmake ..
4. make

## System Information
When posting bug reports, include the following information
- Operating System: Archlinux
- GCC: 11.1
- QGC Version: 4.1.3
- QGC build: building from source
- Flight Controller: N/A
- Autopilot (with version): N/A

## Detailed Description
Provide further details about your issue/bug.

<details>
  <summary>Log</summary>

  ```
In file included from /usr/include/glib-2.0/glib/gthread.h:32,
                 from /usr/include/glib-2.0/glib/gasyncqueue.h:32,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/gstreamer-1.0/gst/gst.h:27,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/gstqtglutility.h:24,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/gstqtglutility.cc:25:
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/gstqtglutility.cc: In function ‘GstGLDisplay* gst_qt_get_gl_display()’:
/usr/include/glib-2.0/glib/gatomic.h:117:19: error: argument 2 of ‘__atomic_load’ must not be a pointer to a ‘volatile’ type
  117 |     __atomic_load (gapg_temp_atomic, &gapg_temp_newval, __ATOMIC_SEQ_CST); \
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gthread.h:260:7: note: in expansion of macro ‘g_atomic_pointer_get’
  260 |     (!g_atomic_pointer_get (location) &&                             \
      |       ^~~~~~~~~~~~~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/gstqtglutility.cc:65:7: note: in expansion of macro ‘g_once_init_enter’
   65 |   if (g_once_init_enter (&_debug)) {
      |       ^~~~~~~~~~~~~~~~~
make: *** [Makefile:134434: gstqtglutility.o] Error 1
make: *** Waiting for unfinished jobs....
In file included from /usr/include/glib-2.0/glib/gthread.h:32,
                 from /usr/include/glib-2.0/glib/gasyncqueue.h:32,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/gstreamer-1.0/gst/gst.h:27,
                 from /usr/include/gstreamer-1.0/gst/video/video.h:23,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/gstqsgtexture.cc:28:
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/gstqsgtexture.cc: In constructor ‘GstQSGTexture::GstQSGTexture()’:
/usr/include/glib-2.0/glib/gatomic.h:117:19: error: argument 2 of ‘__atomic_load’ must not be a pointer to a ‘volatile’ type
  117 |     __atomic_load (gapg_temp_atomic, &gapg_temp_newval, __ATOMIC_SEQ_CST); \
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gthread.h:260:7: note: in expansion of macro ‘g_atomic_pointer_get’
  260 |     (!g_atomic_pointer_get (location) &&                             \
      |       ^~~~~~~~~~~~~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/gstqsgtexture.cc:42:7: note: in expansion of macro ‘g_once_init_enter’
   42 |   if (g_once_init_enter (&_debug)) {
      |       ^~~~~~~~~~~~~~~~~
make: *** [Makefile:134439: gstqsgtexture.o] Error 1
In file included from /usr/include/glib-2.0/glib/gthread.h:32,
                 from /usr/include/glib-2.0/glib/gasyncqueue.h:32,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/gstreamer-1.0/gst/gst.h:27,
                 from /usr/include/gstreamer-1.0/gst/video/video.h:23,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/qtwindow.cc:27:
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/qtwindow.cc: In constructor ‘QtGLWindow::QtGLWindow(QWindow*, QQuickWindow*)’:
/usr/include/glib-2.0/glib/gatomic.h:117:19: error: argument 2 of ‘__atomic_load’ must not be a pointer to a ‘volatile’ type
  117 |     __atomic_load (gapg_temp_atomic, &gapg_temp_newval, __ATOMIC_SEQ_CST); \
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gthread.h:260:7: note: in expansion of macro ‘g_atomic_pointer_get’
  260 |     (!g_atomic_pointer_get (location) &&                             \
      |       ^~~~~~~~~~~~~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/qtwindow.cc:114:7: note: in expansion of macro ‘g_once_init_enter’
  114 |   if (g_once_init_enter (&_debug)) {
      |       ^~~~~~~~~~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/qtwindow.cc: In member function ‘void QtGLWindow::beforeRendering()’:
/usr/include/glib-2.0/glib/gatomic.h:117:19: error: argument 2 of ‘__atomic_load’ must not be a pointer to a ‘volatile’ type
  117 |     __atomic_load (gapg_temp_atomic, &gapg_temp_newval, __ATOMIC_SEQ_CST); \
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gthread.h:260:7: note: in expansion of macro ‘g_atomic_pointer_get’
  260 |     (!g_atomic_pointer_get (location) &&                             \
      |       ^~~~~~~~~~~~~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/qtwindow.cc:160:7: note: in expansion of macro ‘g_once_init_enter’
  160 |   if (g_once_init_enter(&once)) {
      |       ^~~~~~~~~~~~~~~~~
make: *** [Makefile:134458: qtwindow.o] Error 1
In file included from /usr/include/glib-2.0/glib/gthread.h:32,
                 from /usr/include/glib-2.0/glib/gasyncqueue.h:32,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/gstreamer-1.0/gst/gst.h:27,
                 from /usr/include/gstreamer-1.0/gst/video/video.h:23,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/qtitem.cc:27:
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/qtitem.cc: In constructor ‘QtGLVideoItem::QtGLVideoItem()’:
/usr/include/glib-2.0/glib/gatomic.h:117:19: error: argument 2 of ‘__atomic_load’ must not be a pointer to a ‘volatile’ type
  117 |     __atomic_load (gapg_temp_atomic, &gapg_temp_newval, __ATOMIC_SEQ_CST); \
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gthread.h:260:7: note: in expansion of macro ‘g_atomic_pointer_get’
  260 |     (!g_atomic_pointer_get (location) &&                             \
      |       ^~~~~~~~~~~~~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/libs/qmlglsink/gst-plugins-good/ext/qt/qtitem.cc:111:7: note: in expansion of macro ‘g_once_init_enter’
  111 |   if (g_once_init_enter (&_debug)) {
      |       ^~~~~~~~~~~~~~~~~
make: *** [Makefile:134465: qtitem.o] Error 1
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/src/AutoPilotPlugins/PX4/PX4RadioComponent.cc: In member function ‘virtual bool PX4RadioComponent::setupComplete() const’:
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/src/AutoPilotPlugins/PX4/PX4RadioComponent.cc:46:28: error: loop variable ‘mapParam’ of type ‘const QString&’ binds to a temporary constructed from type ‘const char* const’ [-Werror=range-loop-construct]
   46 |         for(const QString &mapParam : {"RC_MAP_ROLL", "RC_MAP_PITCH", "RC_MAP_YAW", "RC_MAP_THROTTLE"}) {
      |                            ^~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/src/AutoPilotPlugins/PX4/PX4RadioComponent.cc:46:28: note: use non-reference type ‘const QString’ to make the copy explicit or ‘const char* const&’ to prevent copying
cc1plus: all warnings being treated as errors
make: *** [Makefile:126584: PX4RadioComponent.o] Error 1
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/src/AutoPilotPlugins/PX4/PX4AdvancedFlightModesController.cc: In member function ‘void PX4AdvancedFlightModesController::_init()’:
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/src/AutoPilotPlugins/PX4/PX4AdvancedFlightModesController.cc:104:25: error: loop variable ‘attitudeParam’ of type ‘const QString&’ binds to a temporary constructed from type ‘const char* const’ [-Werror=range-loop-construct]
  104 |     for (const QString &attitudeParam : {"RC_MAP_THROTTLE", "RC_MAP_YAW", "RC_MAP_PITCH", "RC_MAP_ROLL", "RC_MAP_FLAPS", "RC_MAP_AUX1", "RC_MAP_AUX2"}) {
      |                         ^~~~~~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/src/AutoPilotPlugins/PX4/PX4AdvancedFlightModesController.cc:104:25: note: use non-reference type ‘const QString’ to make the copy explicit or ‘const char* const&’ to prevent copying
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/src/AutoPilotPlugins/PX4/PX4AdvancedFlightModesController.cc: In member function ‘void PX4AdvancedFlightModesController::_validateConfiguration()’:
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/src/AutoPilotPlugins/PX4/PX4AdvancedFlightModesController.cc:173:24: error: loop variable ‘thresholdParam’ of type ‘const QString&’ binds to a temporary constructed from type ‘const char* const’ [-Werror=range-loop-construct]
  173 |     for(const QString &thresholdParam : {"RC_ASSIST_TH", "RC_AUTO_TH", "RC_ACRO_TH", "RC_POSCTL_TH", "RC_LOITER_TH", "RC_RETURN_TH", "RC_OFFB_TH"}) {
      |                        ^~~~~~~~~~~~~~
/home/acxz/vcs/git/github/acxz/pkgbuilds/qgroundcontrol/src/qgroundcontrol-4.1.3/src/AutoPilotPlugins/PX4/PX4AdvancedFlightModesController.cc:173:24: note: use non-reference type ‘const QString’ to make the copy explicit or ‘const char* const&’ to prevent copying
cc1plus: all warnings being treated as errors
  ```
</details>

## Log Files and Screenshots
- [QGC Console Logs](https://docs.qgroundcontrol.com/en/Support/Support.html#reporting-bugs)
- Autopilot logs when available (post a link).
- Screenshots of QGC to help identify the current issue/bug behavior.
N/A

--- a/src/AutoPilotPlugins/PX4/PX4AdvancedFlightModesController.cc
+++ b/src/AutoPilotPlugins/PX4/PX4AdvancedFlightModesController.cc
@@ -101,7 +101,7 @@ void PX4AdvancedFlightModesController::_init(void)
     // Setup the channel combobox model
     QVector<int> usedChannels;
 
-    for (const QString &attitudeParam : {"RC_MAP_THROTTLE", "RC_MAP_YAW", "RC_MAP_PITCH", "RC_MAP_ROLL", "RC_MAP_FLAPS", "RC_MAP_AUX1", "RC_MAP_AUX2"}) {
+    for (const char* const&attitudeParam : {"RC_MAP_THROTTLE", "RC_MAP_YAW", "RC_MAP_PITCH", "RC_MAP_ROLL", "RC_MAP_FLAPS", "RC_MAP_AUX1", "RC_MAP_AUX2"}) {
         int channel = getParameterFact(-1, attitudeParam)->rawValue().toInt();
         if (channel != 0) {
             usedChannels << channel;
@@ -170,7 +170,7 @@ void PX4AdvancedFlightModesController::_validateConfiguration(void)
     }
     
     // Validate thresholds within range
-    for(const QString &thresholdParam : {"RC_ASSIST_TH", "RC_AUTO_TH", "RC_ACRO_TH", "RC_POSCTL_TH", "RC_LOITER_TH", "RC_RETURN_TH", "RC_OFFB_TH"}) {
+    for(const char* const&thresholdParam : {"RC_ASSIST_TH", "RC_AUTO_TH", "RC_ACRO_TH", "RC_POSCTL_TH", "RC_LOITER_TH", "RC_RETURN_TH", "RC_OFFB_TH"}) {
         float threshold = getParameterFact(-1, thresholdParam)->rawValue().toFloat();
         if (threshold < 0.0f || threshold > 1.0f) {
             _validConfiguration = false;

--- a/src/AutoPilotPlugins/PX4/PX4RadioComponent.cc
+++ b/src/AutoPilotPlugins/PX4/PX4RadioComponent.cc
@@ -43,7 +43,7 @@ bool PX4RadioComponent::setupComplete(void) const
     if (_vehicle->parameterManager()->getParameter(-1, "COM_RC_IN_MODE")->rawValue().toInt() != 1) {
         // The best we can do to detect the need for a radio calibration is look for attitude
         // controls to be mapped.
-        for(const QString &mapParam : {"RC_MAP_ROLL", "RC_MAP_PITCH", "RC_MAP_YAW", "RC_MAP_THROTTLE"}) {
+        for(const char* const&mapParam : {"RC_MAP_ROLL", "RC_MAP_PITCH", "RC_MAP_YAW", "RC_MAP_THROTTLE"}) {
             if (_vehicle->parameterManager()->getParameter(FactSystem::defaultComponentId, mapParam)->rawValue().toInt() == 0) {
                 return false;
             }

